src/components/site-form.tsx

```diff
@@ -106,29 +223,77 @@ export const SiteForm = ( {
 	doesPathContainWordPress?: boolean;
 	isPathInputDisabled?: boolean;
 	onSubmit: ( event: FormEvent ) => void;
+	fileForImport?: File | null;
+	setFileForImport?: ( file: File | null ) => void;
+	onFileSelected?: ( file: File ) => void;
+	fileError?: string;
 } ) => {
 	const { __ } = useI18n();
 
+	const [ isAdvancedSettingsVisible, setAdvancedSettingsVisible ] = useState( false );
+
+	const handleAdvancedSettingsClick = () => {
+		setAdvancedSettingsVisible( ! isAdvancedSettingsVisible );
+	};
+
 	return (
 		<form className={ className } onSubmit={ onSubmit }>
 			<div className="flex flex-col gap-6">
 				<label className="flex flex-col gap-1.5 leading-4">
 					<span className="font-semibold">{ __( 'Site name' ) }</span>
 					<TextControlComponent onChange={ setSiteName } value={ siteName }></TextControlComponent>
 				</label>
-				{ onSelectPath && (
-					<label className="flex flex-col gap-1.5 leading-4">
-						<span onClick={ onSelectPath } className="font-semibold">
-							{ __( 'Local path' ) }
-						</span>
-						<FormPathInputComponent
-							isDisabled={ isPathInputDisabled }
-							doesPathContainWordPress={ doesPathContainWordPress }
-							error={ error }
-							value={ sitePath }
-							onClick={ onSelectPath }
-						/>
-					</label>
+				{ setFileForImport && (
+					<>
+						<div className="flex flex-col gap-1.5 leading-4">
+							<label className="font-semibold">
+								{ __( 'Import a backup' ) }
+								<span className="font-normal">{ __( ' (optional)' ) }</span>
+							</label>
+							<span className="text-a8c-gray-50 text-xs">
+								{ createInterpolateElement(
+									__( 'Jetpack and WordPress backups supported. <button>Learn more</button>' ),
+									{
+										button: (
+											<Button
+												variant="link"
+												className="text-xs"
+												onClick={ () => getIpcApi().openURL( STUDIO_DOCS_URL ) }
+											/>
+										),
+									}
+								) }
+							</span>
+							<FormImportComponent
+								placeholder={ __( 'Select or drop a file' ) }
+								value={ fileForImport }
+								onChange={ ( file ) => setFileForImport( file ) }
```

Comment:
You could pass directly the function.

```suggestion
								onChange={ setFileForImport }
```

Reply:
Updated in https://github.com/Automattic/studio/pull/378/commits/5226cac20e816c871c0dfeeb88f527537a64f1b9